src/__tests__/gatsby-node.spec.js

```diff
@@ -66,133 +44,235 @@ describe('sourceNodes', () => {
+      constructor() {
+        super({
+          richTextResolver: (_contentItem, _context) =>
+            '###landing_page_image_section###',
+        });
+      }
     }
-  }
-
-  class Project extends ContentItem {
-    constructor() {
-      super({
-        linkResolver: (_link, _context) => '###projectlink###',
-        richTextResolver: (_contentItem, _context) =>
-          '###project###',
-      });
+
+    class Project extends ContentItem {
+      constructor() {
+        super({
+          linkResolver: (_link, _context) => '###projectlink###',
+          richTextResolver: (_contentItem, _context) =>
+            '###project###',
+        });
+      }
     }
-  }
 
 
-  it('does resolve rich text element', async () => {
-    const fakeRichTextResponseConfig = new Map();
-    fakeRichTextResponseConfig.set(
+    it('does resolve rich text element', async () => {
+      const fakeRichTextResponseConfig = new Map();
+      fakeRichTextResponseConfig.set(
+        /https:\/\/deliver.kenticocloud.com\/.*\/items/,
+        {
+          fakeResponseJson: fakeItemsResponseWithRichTextElement,
+          throwCloudError: false,
+        });
+      fakeRichTextResponseConfig.set(
+        /https:\/\/deliver.kenticocloud.com\/.*\/types/,
+        {
+          fakeResponseJson: fakeTypesResponseWithRichTextElement,
+          throwCloudError: false,
+        });
+      const deliveryClientConfig = {
+        projectId: 'dummyProject',
+        typeResolvers: [
+          new TypeResolver('landing_page_image_section', () =>
+            new LandingPageImageSection()
+          ),
+          new TypeResolver('project', () =>
+            new Project()),
+        ],
+        httpService: new KenticoCloudJsSdkTestHttpService(
+          fakeRichTextResponseConfig
+        ),
+      };
+      const expectedRichTextValue = fakeItemsResponseWithRichTextElement
+        .items
+        .filter((item) => item.system.codename === 'simple_landing_page')[0]
+        .elements
+        .content
+        .value;
+
+      const createNodeMock = jest.fn();
+      const actions = {
+        actions: {
+          createNode: createNodeMock,
+        },
+        createNodeId: dummyCreation.createNodeId,
+      };
+      const pluginConfiguration = {
+        deliveryClientConfig,
+        languageCodenames: ['default'],
+      };
+
+      await sourceNodes(actions, pluginConfiguration);
+
+      const landingPageCallNodeSelection = createNodeMock
+        .mock
+        .calls
+        .filter((call) => {
+          const firstArgument = call[0];
+          return firstArgument.internal.type.startsWith('KenticoCloudItem')
+            && firstArgument.system.codename === 'simple_landing_page';
+        });
+
+      expect(landingPageCallNodeSelection).toHaveLength(1);
+      expect(landingPageCallNodeSelection[0]).toHaveLength(1);
+      const landingPageNode = landingPageCallNodeSelection[0][0];
+
+      expect(landingPageNode)

```

Comment:
It is required to add check that the components and its content (and its rich text) are resolved properly.